dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/TrackerExportController.java

```diff
@@ -0,0 +1,101 @@
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import static org.hisp.dhis.webapi.controller.tracker.TrackerControllerSupport.RESOURCE_PATH;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.hisp.dhis.common.Pager;
+import org.hisp.dhis.dxf2.events.trackedentity.TrackedEntityInstanceService;
+import org.hisp.dhis.trackedentity.TrackedEntityInstanceQueryParams;
+import org.hisp.dhis.tracker.domain.TrackedEntity;
+import org.hisp.dhis.tracker.domain.mapper.TrackedEntityMapper;
+import org.hisp.dhis.tracker.domain.web.PagingWrapper;
+import org.hisp.dhis.webapi.controller.event.TrackedEntityInstanceCriteria;
+import org.hisp.dhis.webapi.controller.event.mapper.TrackedEntityCriteriaMapper;
+import org.hisp.dhis.webapi.service.ContextService;
+import org.hisp.dhis.webapi.service.TrackedEntityInstanceSupportService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequestMapping( value = RESOURCE_PATH )
+@RequiredArgsConstructor
+public class TrackerExportController
+{
+    private final ContextService contextService;
+
+    private final TrackedEntityCriteriaMapper criteriaMapper;
+
+    private final TrackedEntityInstanceService trackedEntityInstanceService;
+
+    private final TrackedEntityMapper trackedEntityMapper;
+
+    private final TrackedEntityInstanceSupportService trackedEntityInstanceSupportService;
+
+    @GetMapping( produces = APPLICATION_JSON_VALUE )
+    PagingWrapper<TrackedEntity> getInstances( TrackedEntityInstanceCriteria criteria )
+    {
+        List<String> fields = contextService.getFieldsFromRequest();
+
+        TrackedEntityInstanceQueryParams queryParams = criteriaMapper.map( criteria );
+
+        Collection<TrackedEntity> trackedEntityInstances = trackedEntityMapper
+            .fromCollection( trackedEntityInstanceService.getTrackedEntityInstances2( queryParams,
```

Comment:
@stian-sandvold I think we can now rename this method to `getTrackedEntityInstances` and remove the `2`?

Reply:
Done, even if some tests need to be rewritten in a different way since some of them won't run on H2 database now.